### PR TITLE
Make Visual Studio examples more version agnostic where possible

### PR DESCRIPTION
**Pull request by willihay**
Signed-off-by: Willow Hayward <17654918+willihay@users.noreply.github.com>

## Change summary

In general, build directory examples for Visual Studio builds should not contain a version reference (example: `build\windows_vs2019`). Project Manager does not use it, and having build examples that do can lead to confusion and duplicate build directories. Instead, use just the platform, such as `build\windows`.

Also, in some places we can say simply "Visual Studio" instead of "Visual Studio 2019". This is important as we support Visual Studio 2022.

Adds a table of common cmake generators to the **Configure and Build** topic.

In CLI examples that use the "Visual Studio 16" generator, add a note that links to the table of common generators.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Review by micronAMZN — APPROVED** [review]
👍

File: content/docs/user-guide/build/configure-and-build.md [hunk 1]
```diff
@@ -64,7 +71,7 @@ The typical CMake command used to configure a project looks like the following:
 {{% tab name="Windows" %}}
 
 ```cmd
-cmake -B build/windows_vs2019 -S . -G "Visual Studio 16" -DLY_3RDPARTY_PATH=<downloadable-packages-directory>
+cmake -B build/windows -S . -G "Visual Studio 16" -DLY_3RDPARTY_PATH=<downloadable-packages-directory>
```
Comment (sptramer) [inline content/docs/user-guide/build/configure-and-build.md:74]: Minor nit that we can handle later if needed - should we have three tabs here, two for Windows with different VS versions? Or should we just update to "Visual Studio 17" in stabilization / at the time of merge?

**Review by sptramer — APPROVED** [review]

File: content/docs/user-guide/build/configure-and-build.md (hunk 1 above)
Comment (willihay) [reply content/docs/user-guide/build/configure-and-build.md:74]: Having 3 tabs is not a bad idea. I'd prefer that. I'll create a GHI for follow up.

Comment (willihay) [reply content/docs/user-guide/build/configure-and-build.md:74]: https://github.com/o3de/o3de.org/issues/2026

**Pull request merged**